### DISCUSSION TITLE
[XLSX-Renderer] How to add images in the worksheet?

### DISCUSSION THREAD

**Discussion by Jedliu** [Siemienik/XToolset#191] [Q&A] [answered]
Thanks for this great project. 

I'm wondering how to add images in the loop when use xlsx-renderer. I didn't see any information in the documents. And I saw you contributed the images for the  [ExcelJs](https://github.com/exceljs/exceljs#images).

Thank you!

**Comment by Jedliu** [comment]
Sorry, I saw this was closed and moved to discussions. Is this not possible yet at this moment? Thanks,

**Reply by Siemienik** [reply]
@Jedliu I use Issues to track to-do list. The discussion is the right place to make a question and respond to them :). I answered you here: https://github.com/Siemienik/XToolset/discussions/191#discussioncomment-1231879

**Comment by Siemienik** [comment, marked as answer]
Hi @Jedliu 
Could I ask you to add the template file which you used?

About images, currently, it is possible to use only static images - it means, that it is able to put an image in a template, and it will be rendered in the output. There is no `ImageCell` that would put dynamic images. 

However, the desired result might be reached in one of two ways:
1. Simpler, with using postprocessing. As the XLSX-Renderer has ExcelJS output, it is possible to use API to put images where you want.

```js
// render without images:
const workbook = await new Renderer().renderFromFile('./report-template.xlsx', reportData);

// POST-PROCESSING:
// add image @see https://github.com/exceljs/exceljs#add-image-to-workbook
const imageId = workbook.addImage({
  buffer: fs.readFileSync('path/to.image.png'),
  extension: 'png',
});

// apply image to cell rage @see https://github.com/exceljs/exceljs#add-image-over-a-range
workbook.worksheets[0].addImage(imageId, 'B2:D6');


```


2. Advanced - write `ImageCell` on your own and extend `CellTemplatePool`. (PR will be welcome here 😄  )


**Additionally, if want you support growing this project it is possible to order a feature: xts@siemienik.com**
[**❤️ Sponsor**](https://github.com/sponsors/Siemienik)

**Reply by Jedliu** [reply]
This could be a possible solution to post-process and add images thereafter. 

Thank you,
